src/ecpool.erl

```diff
@@ -81,11 +81,25 @@ pool_spec(ChildId, Pool, Mod, Opts) ->
 %% @doc Start the pool sup.
 -spec(start_pool(pool_name(), atom(), [option()]) -> {ok, pid()} | {error, term()}).
 start_pool(Pool, Mod, Opts) ->
-    ecpool_pool_sup:start_link(Pool, Mod, Opts).
+    case ecpool_pool_sup:start_link(Pool, Mod, Opts) of
+        {ok, Pid} ->
+            wait_for_workers_started(Pid, fun() ->
+                gen_server:stop(Pid)
+            end);
+        {error, Reason} ->
+            {error, Reason}
+    end.
 
 %% @doc Start the pool supervised by ecpool_sup
 start_sup_pool(Pool, Mod, Opts) ->
```

Comment:
Add a sepc for this function.
I tried to grep the caller in emqx.git, detailed error reasons are matched.
e.g.
https://github.com/emqx/influxdb-client-erl/blob/f129428f07c80ee1df9f6e25584b6de49a127ebb/src/influxdb.erl#L52-L58